### ISSUE TITLE
Pythonic API

### ISSUE THREAD

**Issue by flying-sheep** [willtownes/glmpca-py#7]
This is a straight translation from R. I’d recommend making this pythonic:

- Don’t use mathy variable names or dict parameters. Use expressive names in lower case, especially for the 3rd parameter onwards:

    - X → counts (In scanpy we do rows=observations, cols=features)
    - L → n_components
    - ctl["maxIter"] → iter_max
    - init["factors"] → init_factors
    - ...

    ```py
    def glm_pca(counts, n_components: int, *, fam="poi", iter_max=1000, ...):
    ```

- Best would be to to use the [scipy way](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html) of creating a class that takes all the model parameters and fitting it to the data, but keeping it one function is also fair.
- If you don’t use a class, Best don’t return a dict, but a namedtuple. This has the advantage that elements are easier to access, and it can be unpacked:

    ```py
    factors, loadings = glm_pca(counts, 10, fam="muli")
    ```

**Comment by willtownes** [comment]
these are all good suggestions- thanks!
